docs/integrations/persisting-store-data.md

```diff
@@ -452,6 +452,75 @@ If your app does depends on the persisted state at page load,
 see [_How can I check if my store has been hydrated_](#how-can-i-check-if-my-store-has-been-hydrated)
 in the [FAQ](#faq) section below.
 
+### Usage in Next.js
+
+NextJS uses Server Side Rendering, and it will compare the rendered component on the server with the one rendered on client.
+But since you are using data from browser to change your component, the two renders will differ and Next will throw a warning at you.
+
+The errors usually are:
+
+- Text content does not match server-rendered HTML
+- Hydration failed because the initial UI does not match what was rendered on the server
+- There was an error while hydrating. Because the error happened outside of a Suspense boundary, the entire root will switch to client rendering
+
+To solve these errors, you will need to create a custom hook that makes so that zustand waits a little bit before changing your components.
+
+create a file and put this code:
+
+```typescript
```

Comment:
You can shorten `typescript` to `ts` everywhere.

Reply:
ok will do